foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/databaseaccess/TableExistenceCheck.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 1998, 2018 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+// Contributors:
+//     Oracle - initial API and implementation from Oracle TopLink
+package org.eclipse.persistence.internal.databaseaccess;
+
+import java.util.Vector;
+
+import org.eclipse.persistence.internal.sessions.DatabaseSessionImpl;
+import org.eclipse.persistence.queries.DataReadQuery;
+import org.eclipse.persistence.tools.schemaframework.FieldDefinition;
+import org.eclipse.persistence.tools.schemaframework.TableDefinition;
+
+/**
+ * INTERNAL:
+ * Interface to define method prototype for table existence check.
+ */
+public interface TableExistenceCheck {
```

Comment:
The only reason of added complexity is to avoid copy-pasting  
 ```java
    try {
          session.setLoggingOff(true);
          session.executeQuery(getQuery(table));
          return true;
      } catch (Exception notFound) {
          return false;
      }
```
into each of the overriding methods. So your DB2 change will define just the query itself.

Reply:
I aggree with @dazey3 here, it looks more complex then it should be. I can imagine having 2 methods methods in DBPLatform ie like this:

```java
public boolean checkTableExists(Session session, TableDefinition table, boolean supressLogging) {
    DataReadQuery query = new DataReadQuery(getTableExistsQuery(session, table));
    query.setMaxRows(1);
    boolean logging = session.getLogLevel...
    try {
        session.setLoggingOff(supressLogging);
        result = session.executeQuery(query);
        return result is not empty
    } catch (Throwable notFound) {
        return false;
    } finally {
        session.setLogLevel(logging)...
    }
}
```

and 

```java
/**
 * Return a Query which returns exactly one row if given table exists and either returns no result or throws an exception if given table does not exist.
**/
protected Query getTableExistsQuery(Session session, TableDefinition table) {
    //and default here would be original one
}
```

in this case, platforms which are able to construct the query as defined by javadoc of getTableExistsQuery method would reimplement just this method and those who would need finer control could override checkTableExists method.

WDYT?